### PR TITLE
Ensure HTTP 1XX response codes work properly

### LINKED ISSUE #1337: 1XX response codes should work properly
Making this work for uploads is worth doing. We should do it, being mindful of the consequences for http/2.0.

### PR DESCRIPTION
**Pull request by ameya-pandilwar**
Fixes #1337

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/http/Http1xStream.java [hunk 1]
```diff
@@ -189,7 +190,7 @@ public void writeRequest(Headers headers, String requestLine) throws IOException
             .message(statusLine.message)
             .headers(readHeaders());
 
-        if (statusLine.code != HTTP_CONTINUE) {
+        if (statusLine.code != HTTP_CONTINUE || statusLine.code != HTTP_PROCESSING) {
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/http/Http1xStream.java]: Shouldn't this be `&&` instead of `||`?

**Comment by swankjesse** [comment]
Test case please.

**Comment by dave-r12** [comment]
This one looks almost complete. I'll try to finish it up with a test case.

**Comment by swankjesse** [comment]
Let’s start a new PR from scratch if we pick up this work.

**Pull request closed without merging**